app/src/main/java/no/birkett/quietshare/MainActivity.java

```diff
@@ -122,8 +121,6 @@ private void handleSendClick() {
     }
 
     private void receive() {
-        receiver.setBlocking(5, 0);
```

Comment:
Blocking on a background thread is absolutely the way to go! In fact I tried that on this branch before realizing you had done it already, which should be in the autoreceive commit. For the sake of button presses I figured nonblocking would be nicer.

Quiet is nonblocking by default for historical reasons. The first thing I bootstrapped it on was single threaded, so it would grab samples from the soundcard and then empty the receive queue out. For a client application, you definitely do want blocking.